### DISCUSSION TITLE
Populate table with data

### DISCUSSION THREAD

**Discussion by pythonormen** [zauberzeug/nicegui#388] [Q&A] [answered]
```py
table = ui.table({
    'columnDefs': [
        {'headerName': 'Name', 'field': 'name'},
        {'headerName': 'Age', 'field': 'age'}
    ],
    'rowData': [
        {'name': 'data[0][0]', 'age': data[0][1]},
        {'name': data[1][0],'age': data[1][1]},
    ],
}, theme="ag-theme-alpine").classes('w-full')

def update():
    data = fetchData()
    table.options['rowData'][0]['name'] = data[0][0]
    table.options['rowData'][1]['name'] = data[1][0]
    table.options['rowData'][0]['age'] = data[0][1]
    table.options['rowData'][1]['age'] = data[1][1]
    table.update()

ui.timer(interval=60, callback=update)
```

Hello
I have a simple table that I want to present certain data and refresh the content every minute. Any tip on how I can make this scale to e.g. 50 rows and present the data with a for loop or any other way?
Much appreciated

**Comment by rodja** [comment, marked as answer]
Well it obviously depends on the data you want to fill in. But in general a `for` loop should work.

**Reply by pythonormen** [reply]
Would you mind how it would be inserted under 'rowData'?
(regardless of source)
Can't figure out how.
Thanks in advance

**Reply by falkoschindler** [reply]
```py
for i in range(50):
    table.options['rowData'][i]['name'] = data[i][0]
    table.options['rowData'][i]['age'] = data[i][1]
```

**Reply by pythonormen** [reply]
> ```python
> for i in range(50):
>     table.options['rowData'][i]['name'] = data[i][0]
>     table.options['rowData'][i]['age'] = data[i][1]
> ```


```
'rowData': [
        {'name': 'data[0][0]', 'age': data[0][1]},
        {'name': data[1][0],'age': data[1][1]},
    ],
```
Thanks but how about this section?

**Reply by falkoschindler** [reply]
You can use a list comprehension:

```py
'rowData': [{'name': item[0], 'age': item[1]} for item in data]
```

**Comment by Shahin-rmz** [comment]
Hi, would you please tell me where did you get information about `table` api?
Thanks

**Reply by falkoschindler** [reply]
The AG Grid table @pythonormen is using here is now called `ui.aggrid`. You can find the documentation here: https://nicegui.io/reference#ag_grid
In upcoming NiceGUI version 1.2.0 the `ui.table` implements Quasar's QTable (#500).

**Reply by Shahin-rmz** [reply]
Thanks. it is helpful. Although for importing a simple CSV it looks like a bit overkill.
is there any simpler way to import and show csv as `pandas ` or `polars` DataFrame?

**Reply by Shahin-rmz** [reply]
Here is how I managed to do it, but it's a bit too much I think.

```python
data_dict = {
    'columnDefs': [
        {'headerName': col, 'field': col} for col in df.columns
    ],
    'rowData': df.to_dict('records'),
    'rowSelection': 'multiple',
}
```

**Reply by falkoschindler** [reply]
Yes, something like `ui.aggrid.from_pandas(df)` would be handy. We'll think about it.

**Reply by falkoschindler** [reply]
I created PR #557. Let's continue discussing this feature over there.
